src/Pages/Expenses.cshtml.cs

```diff
@@ -8,12 +10,41 @@ public class ExpensesModel : PageModel
     {
         public IEntryList Expenses { get; set; }
 
+        [BindProperty]
+        public string Title { get; set; }
+        [BindProperty]
+        public string Amount { get; set; }
+        [BindProperty]
+        public string Date { get; set; }
+        [BindProperty]
+        public string Importance { get; set; }
+        [BindProperty]
+        public string IsMonthly { get; set; }
+
+        public string Error { get; set; }
+
         public void OnGet()
         {
+            //ka�kas neveikia paspaudus post, meta kad tu��ias List???
             var dataManager = new DataManager();
             Expenses = dataManager.Expenses.EntryList;
             //var expensesManager = new EntryManager(EntryType.Expense);
             //Expenses = expensesManager.EntryList;
         }
+
+        public void OnPost()
+        {
+            if (!decimal.TryParse(Amount, out var parsedAmount))
+            {
+                Error = "Amount is not a number!";
+                return;
+            }
+
+            var parsedDate = Convert.ToDateTime(Date);
+            var parsedIsMonthly = Convert.ToBoolean(IsMonthly);
+            var parsedImportance = int.Parse(Importance);
+            var Entry = new Entry(Title, parsedAmount,parsedDate,parsedIsMonthly,parsedImportance);
+            //kodas prid�t � listus (b�t� gerai jei Arnas padaryt�)
```

Comment:
To add an Entry, just call Expenses.EntryList.Add(Entry). It returns a boolean if it worked

Reply:
Is this working in current files, or in the new not merged pull request, since this exact line doesn't seem to work ?

Reply:
My mistake, should work with dataManager.Expenses.EntryList.Add(Entry)